warehouse/query-core/src/test/java/datawave/query/iterator/QueryOptionsTest.java

```diff
@@ -259,4 +268,182 @@ protected static String decompressOption(final String buffer, Charset characterS
         }
     }
 
+    @Test
+    public void testDefaultOptions() {
+
+        IteratorSetting iteratorSetting = new IteratorSetting(1, "foo", QueryIterator.class, new HashMap<>());
+        Map<String,String> nonDefaults = new HashMap<>();
+        nonDefaults.put(QueryOptions.INCLUDE_RECORD_ID, "false");
+        nonDefaults.put(QueryOptions.INCLUDE_DATATYPE, "true");
+
+        // Test all default
+        QueryOptions.getDefaultOptions(QueryIterator.class.getName()).getDefaultValues().forEach((k, v) -> {
+            QueryOptions.addOption(iteratorSetting, k, v, false);
+        });
+
+        Assert.assertEquals(0, iteratorSetting.getOptions().size());
+
+        // Test non-defaults added after all defaults added
+        iteratorSetting.clearOptions();
+        QueryOptions.getDefaultOptions(QueryIterator.class.getName()).getDefaultValues().forEach((k, v) -> {
+            QueryOptions.addOption(iteratorSetting, k, v, false);
+        });
+
+        iteratorSetting.addOptions(nonDefaults);
+        assertEquals(2, iteratorSetting.getOptions().size());
+
+        // Test defaults added after non-defaults added
+        iteratorSetting.clearOptions();
+        iteratorSetting.addOptions(nonDefaults);
+        QueryOptions.getDefaultOptions(QueryIterator.class.getName()).getDefaultValues().forEach((k, v) -> {
+            QueryOptions.addOption(iteratorSetting, k, v, false);
+        });
+
+        assertEquals(0, iteratorSetting.getOptions().size());
+
+    }
+
+    @Test
+    public void testCollectionAsValueEquivalence() {
+
+        IteratorSetting iteratorSetting = new IteratorSetting(1, "foo", QueryIterator.class, new HashMap<>());
+
+        HashMap<String,Integer> limitFields = new HashMap<>();
+        limitFields.put("field1", 1);
+        limitFields.put("field2", 2);
+
+        QueryOptions.addOption(iteratorSetting, QueryOptions.LIMIT_FIELDS, limitFields, false);
+
+        Assert.assertTrue(QueryOptions.getDefaultOptions(QueryOptionsWithNonEmptyCollectionsAsDefaultValues.class.getName())
+                        .equalsDefaultValue(QueryOptions.LIMIT_FIELDS, limitFields));
+
+    }
+
```

Comment:
Please wrap this map creation with `// @formatter:off` and  `// @formatter:on`, and ensure each put statement is on a separate line for better readability.